core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/AvroUpsertMonitor.scala

```diff
@@ -446,12 +448,17 @@ class AvroUpsertMonitorActor(val pollInterval: FiniteDuration,
       // convenience method to encapsulate the call to EntityService's batchUpdateEntitiesInternal
       def performUpsertBatch(idx: Long, upsertBatch: Seq[EntityUpdateDefinition]): Future[Traversable[Entity]] = {
         logger.info(s"upserting batch #$idx of ${upsertBatch.size} entities for jobId ${jobId.toString} ...")
+
+        // translate the upsertBatch back into a stream of json.
+        // TODO CORE-427: rewrite this monitor to fully stream, instead of stream->materialize->stream
+        val entityUpdateStream: Source[EntityUpdateDefinition, _] = Source(upsertBatch)
+
```

Comment:
up at line 415, we read a file in a bucket into an fs2 Stream. Then, code in this file maps that to an fs2 Stream of `EntityUpdateDefinition`, batches that stream, and materializes each batch. Then here, we turn the materialized batch back into a stream (this time an akka Source, not an fs2 Stream) and we upsert that batch.

With the changes in this PR, we could simplify this process a whole lot, and avoid any batching. However, I don't want to blow up scope on this PR. I suggest we save that work for a follow-on PR which we can prioritize separately since it doesn't block anything.